### PR TITLE
tests: Use a fake retry scheduler in Storage tests

### PR DESCRIPTION
**Pull request by jskeet**
(I'd like both Amanda and Anshul to review - Anshul as it's relevant to her current work, and Amanda as she's already familiar with our scheduler facilities.)

It's worth reviewing one commit at a time.

### CODE REVIEW

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs [hunk 1]
```diff
@@ -89,41 +90,45 @@ public void ExceptionAfterRetryExhaustedTest()
...
-        }
-        else
-        {
-            // The call throws an exception
-            Assert.Throws<GoogleApiException>(() => client.GetBucket("bucket", new GetBucketOptions { RetryOptions = retryOptions }));
-        }
-
-        for (int i = 0; i < messageHandler.AttemptTimestamps.Count; i++)
+        var client = new StorageClientImpl(service, encryptionKey: null, scheduler);
+        scheduler.Run(() =>
```
Comment (jskeet) [inline apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs:102]: This change is actually very small - it's mostly indentation. All I did was wrap the code in a call to scheduler.Run, and change DateTime.UtcNow to clock.GetCurrentDateTimeUtc()

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs (hunk 1 above)
Comment (anshuldavid13) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs:102]: > @anshuldavid13: When this is in, my _hope_ is that in your other code, you can just change "RetryHandler.MarkAsRetriable" to "MarkAsRetriable" everywhere.
Yes Jon I had already created a PR in my fork for the same. Will update it once this is merged.

**Comment by anshuldavid13** [comment]
LGTM

**Pull request merged**